### ISSUE TITLE
Docker fails everything. Can't start, can't restart, can't reset to factory defaults, can't clean up data.

### ISSUE THREAD

**Issue by GeorgeFlorian** [docker/for-win#7366]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID:
Failed to reset to default: 12BE6FC7-C671-4EC5-8C33-7B8893BD9248/20200622154800
Failed to restart: 12BE6FC7-C671-4EC5-8C33-7B8893BD9248/20200622154924
Failed to cleanup data: 12BE6FC7-C671-4EC5-8C33-7B8893BD9248/20200622155221

### Expected behavior
To work.
### Actual behavior
It doesn't' work.
### Information

  - Windows Version: 10.0.19041 Build 19041
  - Docker Desktop Version: 19.03.8

### Steps to reproduce the behavior

  1. Install Docker
  2. See how it can't do anything.

Yesterday Docker worked. Today I did a fresh install of Windows 10 and then installed Docker and it doesn't work anymore.

I have installed Docker. I have tried running command: `docker run --name my-redis-container -p 6379:6379 -d redis`. This returns:
```
docker: error during connect: Post http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.40/containers/create?name=my-redis-container: open //./pipe/docker_engine: The system cannot find the file specified. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.
See 'docker run --help'.
```
Then I ran `Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All`, restarted my PC and the issue still persists. 

Going to Task Manager -> Performance I can see that Virtualization is Enabled.
Going to Programs and Features -> Turn Windows Features on or off I can see that Hyper-V is fully enabled (checked).

**Comment by sstorie** [comment]
I had a similar issue and needed to remove the Hyper-V feature from Windows, restart to let that take effect, and then re-install Docker and let it handle enabling the feature. No clue why that helped me, but it got me going again.

**Comment by GeorgeFlorian** [comment]
> I had a similar issue and needed to remove the Hyper-V feature from Windows, restart to let that take effect, and then re-install Docker and let it handle enabling the feature. No clue why that helped me, but it got me going again.

I did that. It did not help, unfortunately.

**Comment by sstorie** [comment]
Bummer, best of luck because I can't find much evidence that the recent issues with 2.3.0.3 are being acknowledged or worked on. Hopefully that's just my own ignorance around how development on this project works :)

**Comment by GeorgeFlorian** [comment]
Great repo. No help whatsoever.

**Comment by GeorgeFlorian** [comment]
Is this repository dead ?
This is a critical failure of your software.
I had to install Linux to make it work.

How come one day it worked and the other it didn't ? 
Are no way of knowing what happened, even after I have posted here every available information ?

**Comment by JoshuaJarman** [comment]
i am having the same issue after the latest windows 10 insider update.  docker cannot start.  docker cannot be reset.

**Comment by JoshuaJarman** [comment]
I was able to resolve my issue by running the following from PowerShell:
```
wslconfig /unregister docker-desktop
wslconfig /unregister docker-desktop-data
```
Then restarting Docker allowed me to reset all settings and purge data.
Then restarting Docker again started it in a non-broken state and I could resume working.

Hope that helps anyone else.
the docker wsl images are the issue, unregistered those and letting docker rest them resolved in my case.

**Comment by basvs030** [comment]
I had the same issue. first I tried @JoshuaJarman 's tip about wslconfig but, wslconfig could not find anything so I had no luck with that. 

When I installed Docker, it did not ask me anything about Hyper-V, so I figured there might be an issue there. I then installed Hyper-V using the Powershell command below, and rebooted:
`Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All`

This did not help at all. However, after I deïnstalled Docker, and then installed it again, after a reboot it did work. I managed to install Pi-hole and it is running successfully.
